data_visualization_in_ggplot2/data_visualization_ggplot2.md

```diff
@@ -42,31 +97,84 @@ This module also assumes some basic familiarity with R, including
 
-[![Open Binder environment.](https://binder.pangeo.io/badge_logo.svg)](https://binder.pangeo.io/v2/gh/arcus/education_r_environment/roseh-data-viz-module?urlpath=rstudio) **Click the "launch binder" button!**
+  <a href = "https://mybinder.org/v2/gh/arcus/education_r_environment/main?urlpath=rstudio" target = "_blank"><img src="https://mybinder.org/static/images/badge_logo.svg"></a> **← Click the "launch binder" button!**
+
+<div class = "hint" style = "align-items: center; display: flex;">
+
+<div style = "margin: 1rem; max-width: 45%; float:left;"> If you're the first person to fire up this environment in a while, you might see this loading screen for up to five minutes.  Be patient!</div>
+<div style = "margin: 1rem auto; max-width: 45%; float:left;"> ![Binder loading screen.](media/binder_loading.gif)<!--
+style = "border: 1px solid rgb(var(--color-highlight));"-->
+</div>
+</div>
+
+**Then**, once you have access to RStudio and you see something like the image below, you'll need to open the sample data for this course.  In the file area to the lower right, you'll see, among multiple choices, the folder called "@r_code".  That's the code for this module!
+
+![RStudio as shown in the cloud platform Binder.](media/binder_rstudio.png)<!--
+style = "border: 1px solid rgb(var(--color-highlight)); max-width: 800px;"-->
+
+<h3>Option 2: Work on Your Computer</h3>
+
+If you have [R](https://www.r-project.org/) and [RStudio](https://www.rstudio.com/products/rstudio/download/#download) installed already on your local computer, you might be interested in simply downloading our sample code to your computer. Here's how.  Note -- if you've already done this step in another module, you might have the material for this module already!
+
+* In RStudio, open a new project (File, New Project)
+* Select Version Control, then Git
+* Drop this link into the "Repository URL": https://github.com/arcus/education_r_environment
+* Change the "Project directory name" and "Create project as a subdirectory of" boxes to suit your needs (where will this code be stored on your computer?).
+* Click to select the "Open in new session" checkbox
+* Click "Create Project"
+* In the file area to the lower right, you'll see, among multiple choices, the folder called "@r_code".  That's the code for this module!
+
+**Want to watch this process?  Click on the image below to play an animated gif.  It will continue to loop and you can re-start it by clicking again.**
+
+<div style="display:none">@gifPreload</div>
```

Comment:
This GIF isn't rendering, but I think the extra line in the YAML should make it work.

Reply:
Or possibly it's because the url is pointing to main?

Reply:
Yeah, I pointed it to what will eventually be the correct link, but it means it doesn't render properly now...